modules/services/model/src/main/java/au/com/shiftyjelly/pocketcasts/models/entity/BaseEpisode.kt

```diff
@@ -41,6 +41,7 @@ sealed interface BaseEpisode {
     var playedUpToModified: Long?
     var playErrorDetails: String?
     var downloadErrorDetails: String?
+    var deselectedChapters: String?
```

Comment:
I agree with @MiSikora for this approach. I would say, We should further scale it to more simpler approach. In @MiSikora approach, It will be hard to remove or adding elements directly using SQL query and will lead to decrease in performance. 

Instead, We should create a separate table having for tracking the selection or deselection under episode_id, chapter_id keys in table as composition primary key as Room supports this.

Reply:
> It will be hard to remove or adding elements directly using SQL query

This part is true. Without database normalization we won't be able to access things directly through SQL statements. I don't know if it is something that we need for deselected chapters. Chapters themselves are already baked into file metadata and aren't stored in our database. It's hard for me to imagine a scenario where we need to use direct queries only on chapters and not on episodes.

> and will lead to decrease in performance.

This is hard to tell without benchmarks. Pocket Casts conflated storage and domain models. Having a separate table requires more queries to execute when we want to read episode metadata. Same thing goes for updates.

---

Having said that I'm fine with this solution as well if you @ashiagr prefer to implement it. But it is not without its cons.

Reply:
Theirs a feature that I'm requesting for this project that RSS time tags will be converted to chapters, mainly because I see them more then embedded chapters. Hopefully it can be added easily enough 🤔🤔

Reply:
I went ahead with a simple conversion from `String` to `ChapterIndices` for current purposes. It wouldn't be hard to update the structure when the need comes.

Thanks for all the comments. I appreciate it. 🙂